File: llvm/shift_test.c
void put(char c,char* ptr){
  *ptr=c;
}

void show(char c){
  short* p_num=(char*)32768;
  short num=*p_num;
  char* location=(char*)(32770+num);

  put(c,location);
  (*p_num)++;
}

void show_num(unsigned char num){
  show((num/((unsigned char)100))+'0');
  show((num%((unsigned char)100)/((unsigned char)10))+'0');
  show((num%((unsigned char)10))+'0');
}


void test(){
  //long i=1;
  //i=i<<17;
  //short a=i>>16;
  //short b=i&0xffff;
	show_num(12);
	show_num(123);
}

File: llvm/shift_test.rs
#[no_mangle]
fn put(c:i8,p_c:*mut i8){
    unsafe{
        *p_c=c;
    }
}
#[no_mangle]
fn show(c:i8){
  let p_num=32768u16 as *mut u16;
  let mut num:u16=unsafe{*p_num};
  let location=(32770u16 + num) as *mut i8;

  put(c,location);
  unsafe{*p_num+=1u16;};
}